packages/@react-spectrum/tag/src/Tag.tsx

```diff
@@ -10,82 +10,92 @@
+  let labelRef = useRef();
+  let tagRef = useRef();
+  let tagRowRef = useRef();
+  let {clearButtonProps, labelProps, tagProps, tagRowProps} = useTag({
+    ...props,
+    isDisabled,
+    isFocused,
+    isRemovable,
+    item,
     onRemove,
-    role
-  } =  useTagGroupProvider();
+    tagRef,
+    tagRowRef
+  }, state);
 
-  let removable = isGroupRemovable !== undefined ? isGroupRemovable : isRemovable;
-  let disabled = isGroupDisabled !== undefined ? isGroupDisabled : isDisabled;
-  let isInvalid = (validationState !== undefined ? validationState : groupValidationState) === 'invalid';
-  let {clearButtonProps, labelProps, tagProps} = useTag({
-    ...props,
-    isRemovable: removable,
-    isDisabled: disabled,
-    validationState: validationState !== undefined ? validationState : groupValidationState,
-    onRemove: props.onRemove || onRemove,
-    role
-  });
-  let {role: buttonRole, ...otherButtonProps} = clearButtonProps;
-  let icon = props.icon || (isInvalid && <Alert />);
+  console.log(children);
 
   return (
-    <FocusRing focusRingClass={classNames(styles, 'focus-ring')}>
+    <div
+      {...tagRowProps}>
       <div
-        {...styleProps}
-        {...mergeProps(tagProps, hoverProps)}
+        {...mergeProps(tagProps, hoverProps, focusProps)}
+        role="gridcell"
         className={classNames(
           styles,
           'spectrum-Tags-item',
           {
-            'is-disabled': disabled,
-            // 'is-selected': isSelected,
-            'spectrum-Tags-item--removable': removable,
-            'is-invalid': isInvalid,
+            'is-disabled': isDisabled,
+            'focus-ring': isFocusVisible,
+            'is-focused': isFocused,
+            'not-removable': !isRemovable,
             'is-hovered': isHovered
           },
           styleProps.className
         )}>
-        {icon && React.cloneElement(icon, {size: 'S', UNSAFE_className: classNames(styles, 'spectrum-Tags-itemIcon')})}
-        <span
-          {...labelProps}
-          className={classNames(styles, 'spectrum-Tags-itemLabel')}>
-          {props.children}
-        </span>
-        {removable &&
-          <span role={buttonRole}>
-            <ClearButton
-              tabIndex={tagProps.tabIndex}
-              focusClassName={classNames(styles, 'is-focused')}
-              UNSAFE_className={classNames(styles, 'spectrum-Tags-itemClearButton')}
-              {...otherButtonProps} />
-          </span>
-        }
+        <SlotProvider
+          slots={{
+            tagLabel: {UNSAFE_className: classNames(styles, 'spectrum-Tag-label')},
+            icon: {UNSAFE_className: classNames(styles, 'spectrum-Tag-icon'), size: 'XS'},
+            text: {UNSAFE_className: classNames(styles, 'spectrum-Tag-content', {'tags-removable': isRemovable})}
+          }}>
+          {typeof children === 'string' ? <div ref={labelRef} {...labelProps}><Text>{children}</Text></div> : children}
```

Comment:
isn't this fundamentally different from what a user would pass in? I think it should be
```suggestion
          {typeof children === 'string' ? <Text>{children}</Text> : children}
```
where labelRef and labelProps are passed on the `text` slot? though what is the labelRef actually used for? i don't see it passed anywhere, looks like it's created and assigned, nothing else.

Reply:
The `labelProps` has the `labelId` which in this case is used for voiceover in the remove scenario. I don't removing `labelRef` and any other unused refs.

Reply:
I think we need to remove that extra div as well though, it's not something that the user would provide, which means that providing a string vs providing React element children would be different right?

```
<Tag>apples</Tag>
```
would be labelled differently than
```
<Tag><Text>apples</Text></Tag>
```
Or am I reading into this the wrong way?

Reply:
you're right. i didn't consider that when adding `labelProps`. i might have to restyle the `children` scenario

Reply:
Will address in followup PR